File: ZoneSentry/Models/RentRequest.cs
namespace ZoneSentry.Models;

public class RentRequest
{
    public int Id { get; set; }
    
    public int PaymentPerMonth { get; set; } 
    public int DurationInMonths { get; set; }
    
    public Realty Realty { get; set; }
    public ApplicationUser Tenant { get; set; }
}

public class RentRequestDTO
{
    public int Id { get; set; }
    
    public int PaymentPerMonth { get; set; } 
    public int DurationInMonths { get; set; }
    
    public RealtyUserView Realty { get; set; }
    public ApplicationUserDTO Tenant { get; set; }
}

public class RentRequestCreate
{
    public int DurationInMonths { get; set; }
    
    public int RealtyId { get; set; }
}

File: ZoneSentry/ClientApp/src/api/models/RentRequest.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { ApplicationUser } from './ApplicationUser';
import type { Realty } from './Realty';

export type RentRequest = {
    id?: number;
    paymentPerMonth?: number;
    durationInMonths?: number;
    realty?: Realty;
    tenant?: ApplicationUser;
};